code/controllers/subsystems/dbcore.dm

```diff
@@ -0,0 +1,554 @@
+
+/// Helper proc for handling queued new queries
+/datum/controller/subsystem/dbcore/proc/create_active_query(datum/db_query/query)
+	PRIVATE_PROC(TRUE)
+	SHOULD_NOT_SLEEP(TRUE)
+	//if(IsAdminAdvancedProcCall()) //Disabled because not present in codebase, currently OOS for implementation
+	//	return FALSE
+	run_query(query)
+	queries_active_num++
+	queries_active += query
+	return query
+
+/datum/controller/subsystem/dbcore/proc/process_query(datum/db_query/query)
+	PRIVATE_PROC(TRUE)
+	SHOULD_NOT_SLEEP(TRUE)
+	//if(IsAdminAdvancedProcCall())
+	//	return FALSE
+	if(QDELETED(query))
+		return FALSE
+	if(query.Process(wait))
+		queries_active -= query
+		return FALSE
+	return TRUE
+
+/datum/controller/subsystem/dbcore/proc/run_query_sync(datum/db_query/query)
+	//if(IsAdminAdvancedProcCall())
+	//	return
+	run_query(query)
+	UNTIL(query.Process())
+	return query
+
+/datum/controller/subsystem/dbcore/proc/run_query(datum/db_query/query)
+	//if(IsAdminAdvancedProcCall())
+	//	return
+	query.job_id = rustg_sql_query_async(connection, query.sql, json_encode(query.arguments))
+
+/datum/controller/subsystem/dbcore/proc/queue_query(datum/db_query/query)
+	//if(IsAdminAdvancedProcCall())
+	//	return
+	queries_standby_num++
+	queries_standby |= query
+
+/datum/controller/subsystem/dbcore/Recover()
+	connection = SSdbcore.connection
+
+/datum/controller/subsystem/dbcore/Shutdown()
+	//This is as close as we can get to the true round end before Disconnect() without changing where it's called, defeating the reason this is a subsystem
+	if(SSdbcore.Connect())
+		for(var/datum/db_query/query in queries_current)
+			run_query(query)
+
+		var/datum/db_query/query_round_shutdown = SSdbcore.NewQuery(
+			"UPDATE [format_table_name("round")] SET shutdown_datetime = Now(), end_state = :end_state WHERE id = :round_id",
+			list("end_state" = SSticker.end_game_state, "round_id" = game_id)
+		)
+		query_round_shutdown.Execute()
+		qdel(query_round_shutdown)
+	if(IsConnected())
+		Disconnect()
+
+// Implementation OOS
+// /datum/controller/subsystem/dbcore/can_vv_get(var_name)
+// 	if(var_name == NAMEOF(src, connection))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, all_queries))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_new))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_standby))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+
+// 	return ..()
+
+// /datum/controller/subsystem/dbcore/vv_edit_var(var_name, var_value)
+// 	if(var_name == NAMEOF(src, connection))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, all_queries))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_new))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_standby))
+// 		return FALSE
+// 	if(var_name == NAMEOF(src, queries_active))
+// 		return FALSE
+// 	return ..()
```

Comment:
Any reason this is kept as commented instead of deleted?

Reply:
It will be implemented in the future, but currently out of scope since I don't want to add all the functions required for this.